doc: Describe list arguments to submodule (#199281)
NixOS/nixpkgs: nixos/doc/manual/development/option-types.section.md

Submodules are detailed in [Submodule](#section-option-types-submodule).

`types.submodule` *`module`*

>     This type is equivalent to

Reviewer: I think this sentence should come after the special `:` syntax that seems to mark the start of the definition, but also perhaps also move it one sentence later? Between here:

> to `imports`). Submodules can

Submodules are detailed in [Submodule](#section-option-types-submodule).

`types.submodule` *`module`*

:   A nested module evaluation, containing its own set of options.
    `module` can either be a single module (in the form of an attribute

>     set, a function, or a path), or as a list of modules (which then

Reviewer 2: ```suggestion
    set, a function, or a path), or a list of modules (which then
```

    gets processed just like a list assigned to `imports`). This type is
    equivalent to
    `types.submoduleWith { modules = toList o; shorthandOnlyDefinesConfig = true; }`.
    Submodules can be used in composed types to create multiple

>     occurrences of the submodule. 

Reviewer 2: I don't understand this sentence.

Reviewer 2: Suggestion:

> Submodules are useful in composed types, e.g. `attrsOf (submodule ...)`.

"Multiple occurrences" and "modular options" are vague, a code sample clearly communicates the purpose here.

Submodules are detailed in [Submodule](#section-option-types-submodule).

`types.submodule` *`module`*

:   A nested module evaluation, containing its own set of options.
    `module` can either be a single module (in the form of an attribute
    set, a function, or a path), or as a list of modules (which then
    gets processed just like a list assigned to `imports`). Submodules can be

>     used in composed types to create modular options. This type is

Reviewer: ```suggestion
    used in [composed types](#sec-option-types-composed) to create modular options. This type is
```

Reviewer: "Modular options" seems pleonastic in the context of the module system.
```suggestion
    used in [composed types](#sec-option-types-composed) to create
    multiple occurrences of the submodule. This type is
```

>     equivalent to

Reviewer: The equivalence is an intrinsic property of submodules whereas the preceding sentence is about usage. Could you flip those around?

Author: Do you mean to first write that it is the same as `submoduleWith ...` and then describe the usage?

`types.submodule` *`o`*

:   A set of sub options *`o`*. *`o`* can be an attribute set, a function
    returning an attribute set, a list of attribute sets, or a path to a
    file containing such a value. When the argument is a list, it is
    equivalent to the argument being a set with the argument as the value
    of the `imports` attribute. Submodules are used in composed types to

>     create modular options. This is equivalent to

Reviewer 3: I think this entire paragraph warrants a rewrite, here's my suggestion:

> `types.submodule` *`module`*: A nested module evaluation, containing its own set of options. `module` can either be a single module (in the form of an attribute set, a function, or a path), or a list of modules (which then gets processed just like a list of `imports`). Submodules can be used in composed types to create modular options. This type is equivalent to

Author: I agree that reads better. For me as a newcomer it seems hard to tell whether "a list of `imports`" refers to `imports` or `import`. Maybe this could be clarified using
> which then gets processed just like a list assigned to `imports`
